unittests/CppInterOp/DynamicLibraryManagerTest.cpp

```diff
@@ -54,3 +59,30 @@ TEST(DynamicLibraryManagerTest, Sanity) {
   // invalidated...
   // EXPECT_FALSE(Cpp::GetFunctionAddress("ret_zero"));
 }
+
+TEST(UtilsPlatform, DLTest) {
+  std::string err = "";
+  // CPPINTEROP_LIB_TestSharedLib_DIR_PREFIX specified by cmake though target
```

Comment:
Is that supposed to compute the full path?

Reply:
> Is that supposed to compute the full path?

That depends what `${CMAKE_BINARY_DIR}/unittests/bin/$<CONFIG>/` gives, we can assume that it gives the full path.

Reply:
Why is this necessary? Can't we get the fullpath to the test executable and find back the library?

Reply:
I tried to do it but it seemed to vary across arch and platform and configurations.

Reply:
If you do not want macros in that we can have a config.h.in file and have cmake configure it.

Reply:
I am trying to avoid that but not at the cost of introducing `config.h.in`. I was wondering if we should not try to look for these files in a few places. Like check if the file exists in these couple of places and pass it to the system.

Reply:
> I am trying to avoid that but not at the cost of introducing `config.h.in`. I was wondering if we should not try to look for these files in a few places. Like check if the file exists in these couple of places and pass it to the system.

We can have a small utility program that uses c++ 17's filesystem and searches for it. Would that work?

Reply:
I agree, however, no need to C++ 17 to implement that.

Reply:
Alrighty, I shall do it for POSIX and windows separately.

Reply:
You can use llvm facilities from llvm::fs and llvm::sys that would avoid this.